### PR TITLE
Add webpack bundle analyzer as dev dependency

### PR DESCRIPTION
**Pull request by starsirius**
Similar to https://github.com/artsy/volt/pull/3512, this adds [webpack-bundle-analyzer](https://github.com/webpack-contrib/webpack-bundle-analyzer) to dev dependencies for visualizing webpack bundles and identifying optimization opportunities. From the result, there are definitely room for improvement.

I also adds 2 scripts for easier generation of the visualization.

- To generate visualization for development environment, run
  ```
  yarn bundle:stats && yarn bundle:analysis
  ```
  Screen Shot 2020-07-18 at 10 10 00 PM

- To generate visualization for production environment, run
  ```
  RAILS_ENV=production NODE_ENV=production yarn bundle:stats && yarn bundle:analysis
  ```
  Screen Shot 2020-07-18 at 10 11 53 PM

### CODE REVIEW

File: yarn.lock [hunk 1]
```diff
@@ -4394,7 +4419,7 @@ expect@^26.0.1:
     jest-message-util "^26.0.1"
     jest-regex-util "^26.0.0"
 
-express@^4.17.1:
+express@^4.16.3, express@^4.17.1:
   version "4.17.1"
   resolved "https://registry.yarnpkg.com/express/-/express-4.17.1.tgz#4491fc38605cf51f8629d39c2b5d026f98a4c134"
   integrity sha512-mHJ9O79RqluphRrcw2X/GTh3k9tVv8YcoyY4Kkh4WDMUYKRZUq0h1o0w2rrrxBqM7VoeUVqgb27xlEMXTnYt4g==
```
Comment (joeyAghion) [inline yarn.lock:4425]: It's a little weird that we're getting 2 versions of this now, right?

File: yarn.lock [hunk 2]
```diff
@@ -11280,7 +11352,7 @@ write@1.0.3:
   dependencies:
     mkdirp "^0.5.1"
 
-ws@^6.2.1:
+ws@^6.0.0, ws@^6.2.1:
```
Comment (joeyAghion) [inline yarn.lock:11355]: This too?

File: yarn.lock (hunk 1 above)
Comment (starsirius) [reply yarn.lock:4425]: I think this line just lists out all dependency requirements, in which `express@^4.16.3` is required by webpack-bundle-analyzer:
```
express@^4.16.3, express@^4.17.1:
```
The next line shows the actual version installed (which satisfied both requirements):
``` 
  version "4.17.1"
```

File: yarn.lock (hunk 2 above)
Comment (starsirius) [reply yarn.lock:11355]: Similar to https://github.com/artsy/horizon/pull/160/files#r457366183.

File: yarn.lock (hunk 1 above)
Comment (joeyAghion) [reply yarn.lock:4425]: Ah 👍

**Pull request merged**